### DISCUSSION TITLE
What are the save-txt values?

### DISCUSSION THREAD

**Discussion by philippneugebauer** [ultralytics/yolov5#2032] [Q&A] [answered]
What do the values of the result txt stand for?
The first is the label id and the four others are related to the bounding boxes, but what's their value exactly?

```
1 0.540104 0.296296 0.0489583 0.0925926
1 0.728906 0.187963 0.0192708 0.0314815
```

**Comment by glenn-jocher** [comment, marked as answer]
@philippneugebauer these values are in YOLO label format. See https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data#2-create-labels for details.

### 2. Create Labels

After using a tool like [CVAT](https://github.com/opencv/cvat), [makesense.ai](https://www.makesense.ai/) or [Labelbox](https://labelbox.com/)  to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<img width="800" alt="Image Labels" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg">

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<img width="475" src="https://user-images.githubusercontent.com/26833433/98809572-0bc4d580-241e-11eb-844e-eee756f878c2.png">

**Reply by jaideep11061982** [reply]
@glenn-jocher  
I am bit confused about format of output bbx that is saved as txt
as per below code it is in different format. 
Can you please confirm if bbx are saved in  xcycwh  format or xyxy format. 


`save_txt:  # Write to file                        xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh                        line = (cls, *xywh, conf) if save_conf else (cls, *xywh)  # label format                        with open(txt_path + '.txt', 'a') as f:                            f.write(('%g ' * len(line)).rstrip() % line + '\n')`

How i can denormalize if my image shape is x,y,c

**Reply by glenn-jocher** [reply]
@jaideep11061982 --save-txt uses the exact same format as the above zidane.jpg example.

**Reply by harish-pushan** [reply]
hey guys I am really new to object-detection itself and I just picked up this model I have doubt where does this save-txt save the file

**Comment by greenyjwk** [comment]
Hello @glenn-jocher 

Thank you for the answer. I have a quick question regarding saved texts(using --save-txt) and the confusion matrix. I counted all the rows(bounding boxes) throughout all the text files under /labels. However, the total number of bounding boxes(the total number of rows) is less than the total sum of elements in the confusion matrix(TP + FP + FN). 

I checked if the confusion matrix has a more strict NMS, but it doesn't look like it. By any chance, do you know why the total number of predictions (bounding box) from the confusion matrix is less than the total number of bounding boxes from txt files(under labels directory)

Thank you.
